detect windows platform and download associated zip

edsilv opened the pull request.

edsilv commented:
The path that led me here:

```
$ npm install orbit-db ipfs-api --save

> ipfs-log@1.5.1 postinstall C:\Users\edsilv\github\edsilv\orbit-db-test\node_modules\ipfs-log
> ./scripts/post_install.sh

'.' is not recognized as an internal or external command,
operable program or batch file.
```

I then tried to separately npm install ipfs-log in order to test it and got:

```
C:\Users\edsilv\github\edsilv\ipfs-log\node_modules\go-ipfs-dep\src\index.js:42
        throw new Error('no binary available for os:' + goenv.GOOS)
```

This PR allows me to download the correct binary.

I then started looking at https://github.com/haadcode/ipfs-log/blob/master/scripts/post_install.sh

But it seems like the directories it's deleting are not in node_modules anyway - perhaps npm has changed since?

 @haadcode

haadcode commented:
@edsilv Thank you for the detective work and figuring out a solution for this! Windows hasn't gotten much love in general and I think it'd be a good time to start putting effort to fix this, so this PR would be of huge value to get people using IPFS on windows.

@diasdavid the PR LGTM, can you take a look and merge if you're ok with this?

diasdavid commented on src/index.js:
```diff
@@ -1,38 +1,57 @@
+  installPath = path.resolve(__dirname, '..')
 
-  request
+  if (isUnixy) {
```
Let's call it `isWindows` instead and make it the function instead. Then break it to a separate file

diasdavid commented on src/index.js:
```diff
@@ -1,38 +1,57 @@
+      .on('close', callback))
+  }
 
   function checkPlatform (goenv) {
```
instead of passing the goenv, let's pass the goenv.GOOS directly, so that it is not that var env dependent.

diasdavid commented on src/index.js:
```diff
@@ -1,38 +1,57 @@
+    .get(url)
+    .pipe(
+      unzip.Extract({ path: installPath })
+      .on('close', callback))
```
on 'close' or 'finish'?

diasdavid commented on src/index.js:
```diff
@@ -1,38 +1,57 @@
+    .pipe(
+      unzip.Extract({ path: installPath })
+      .on('close', callback))
+  }
```
you can reuse the request.get(url) by preparing the unzipper/untar stream.

diasdavid commented on src/index.js:
```diff
@@ -1,38 +1,57 @@
 module.exports = function (callback) {
   callback = callback || noop
+
+  var fileName, fileExtension, url, installPath, isUnixy
```
a line for each var assignment

diasdavid commented:
Just CR'ed, apology for the delay.

While you are at it, could you add a travis.yml and a circle.yml? Here is the template:

- https://github.com/ipfs/js-ipfs/blob/master/.travis.yml
- https://github.com/ipfs/js-ipfs/blob/master/circle.yml

Thank you!

edsilv commented:
I removed `-npm run coverage` from .travis.yml as there currently aren't any mocha tests and I think this might have been causing an `Error in plugin 'gulp-spawn-mocha'` that I was seeing.
I also removed `make test` - I looked at the js-ipfs Makefile, but I'm not really sure what's happening there...

mateon1 commented on src/checkPlatform.js:
```diff
@@ -0,0 +1,21 @@
+        return false
+    }
+  },
+  isWindows: function (os) {
```
How about `return os === 'windows'`?

edsilv replied:
done

diasdavid commented on src/index.js:
```diff
@@ -1,52 +1,54 @@
-    .pipe(gunzip())
+  installPath = path.resolve(__dirname, '..')
+
+  fileStream = request.get(url)
```
Since fileStream doesn't change, let's do `const fileStream` instead.

Apologies for going super granular in review, I understand where the pattern of declaring vars at the top comes form, but in JS, things get declared when they are needed and now with ES6, it is expected that things that won't change its pointer be defined with const.

diasdavid replied:
This applies to all the other variables.

edsilv replied:
No worries, using const is definitely better. Added commit

diasdavid commented:
Other than https://github.com/diasdavid/go-ipfs-dep/pull/9#discussion_r83625142, LGTM.

@haadcode I remember that you told me that you also made a change to make go-ipfs-dep be able to pick dev versions, what do you think of this PR?

diasdavid commented:
Thank you! :)

+ go-ipfs-dep@0.4.3-2 published

haadcode commented:
@diasdavid this is the change I mentioned some time ago, haven't PR'd it yet but if you want, go ahead and add it to your repo: https://github.com/haadcode/go-ipfs-dep/commit/aee00e33d97b848166c768b5999e848b472d8a45 (can't do it this week myself).
